contoso_manufacturing/developer/webapp-decode/yolov8.py

```diff
@@ -156,22 +185,41 @@ def draw_detections(self, img, box, score, class_id):
         cv2.putText(img, label, (label_x, label_y), cv2.FONT_HERSHEY_SIMPLEX, 1.2, (0, 0, 0), 1, cv2.LINE_AA)
 
     def run(self):
-        if(self.verbose):
-            print("Running detection...")
+        while not self.stopped:
+            if(self.verbose):
+                print("Running detection...")
+
+            self.frame_number += 1
+            # If mod = 0, i will get the frame and skip it
+            if self.frame_number % self.skip_rate == 0:
+                if not self.frame_queue.empty():
+                    self.cap = self.frame_queue.get()
```

Comment:
I don't think we need to assign the retrieved frame to self.cap here since we are just discarding it?